scripts/GetClrDbg.sh

```diff
@@ -76,97 +115,271 @@ generate_nuget_config()
+
+            if [ "$wcOutput" -gt 0 ]; then
+                echo "Error: files are being used in location '$__InstallLocation'"
+                exit 1
+            fi
+
+            rm -rf $__InstallLocation
+            if [ "$?" -ne 0 ]; then
+                echo "Error: files could not be removed from '$__InstallLocation'"
+                exit 1
+            fi
+        fi
+        echo "Info: Removed directory '$__InstallLocation'"
     fi
-    __InstallPath=$2
-fi
+}
 
-pushd $__InstallPath > /dev/null 2>&1
-if [ "$?" -ne 0 ]; then
-    echo "Error: Unable to cd to install directory '$__InstallPath'"
+# Checks if the existing copy is the latest version.
+check_latest()
+{
+    __SuccessFile="$__InstallLocation/success.txt" 
+    if [ -f "$__SuccessFile" ]; then
+        __LastInstalled=$(cat "$__SuccessFile")
+        echo "Info: LastIntalled version of clrdbg is '$__LastInstalled'"
+        if [ "$__ClrDbgVersion"="__LastInstalled" ]; then
+            __SkipDownloads=true
+            echo "Info: ClrDbg is upto date"
+        fi
+    else
+        echo "Info: Previous installation at "$__InstallLocation" not found"
+    fi
+}
+
+if [ -z "$1" ]; then
+    print_help
     exit 1
+else
+    parse_and_get_arguments $@
+
+    if [ -z $__ClrDbgMetaVersion ]; then
+        get_legacy_arguments $@
+    fi    
 fi
 
-# For the rest of this script we can assume the working directory is the install path
+convert_install_path_to_absolute
+print_and_verify_arguments
+set_clrdbg_version "$__ClrDbgMetaVersion"
+echo "Info: Using clrdbg version '$__ClrDbgVersion'"
 
-echo 'Info: Determinig Runtime ID'
-__RuntimeID=
-get_dotnet_runtime_id
-if [ -z $__RuntimeID ]; then
-    echo "Error: Unable to determine dotnet Runtime ID"
-    echo "GetClrDbg.sh requires .NET CLI Tools version >= 1.0.0-beta-002173. Please make sure your install of .NET CLI is up to date"
-    exit 1
-fi
-echo "Info: Using Runtime ID '$__RuntimeID'"
-
-echo 'Info: Generating project.json'
-generate_project_json $rid
-
-echo 'Info: Generating NuGet.config'
-generate_nuget_config
-
-# dotnet restore and publish add color to their output
-# I have found that the extra characters to provide color can corrupt
-# the shell output when running this as part of docker build
-# Therefore, I redirect the output of these commands to a log
-echo 'Info: Executing dotnet restore'
-dotnet restore > dotnet_restore.log 2>&1
-if [ $? -ne 0 ]; then
-    echo "dotnet restore failed"
-    exit 1
-fi
+process_removal
+check_latest
 
-echo 'Info: Executing dotnet publish'
-dotnet publish -o . > dotnet_publish.log 2>&1
-if [ $? -ne 0 ]; then
-    echo "dotnet publish failed"
-    exit 1
+
+if [ "$__SkipDownloads" = true ]; then
+    echo "Info: Skipping downloads"
+else
+    prepare_install_location
+    pushd $__InstallLocation > /dev/null 2>&1
+    if [ "$?" -ne 0 ]; then
+        echo "Error: Unable to cd to install directory '$__InstallLocation'"
+        exit 1
+    fi
```

Comment:
>fi [](start = 4, length = 2)

In upgrade scenarios, do we want to blow away all the old files before we attempt to download the new ones?